test/common/upstream/health_checker_impl_test.cc

```diff
@@ -2690,6 +2703,54 @@ TEST_F(GrpcHealthCheckerImplTest, Success) {
   expectHostHealthy(true);
 }
 
+// Test single host check success with custom authority.
+TEST_F(GrpcHealthCheckerImplTest, SuccessWithCustomAuthority) {
```

Comment:
This test is fine. However, do you think it is better to have a helper that wraps testing on custom service name and authority value (including the setup function one)? And can we have an expectation to check the header value of `:authority` is the name of the cluster when it is not set?

Reply:
Yes. 

Thanks!